### ISSUE TITLE
When creating/saving a GKE cluster validation for nodepools with same name does not happen

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#34049]
If this is a UI issue, please file an issue at https://github.com/rancher/dashboard/

**Rancher Server Setup**
- Rancher version: 2.6-head commit id: `0ab00d7`
- Installation option (Docker install/Helm Chart): Helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE1
- Proxy/Cert Details: Rancher generated

**Information about the Cluster**
- Kubernetes version: 1.20
- Cluster Type (Local/Downstream): Downstream - GKE v2

**Describe the bug**
When creating/saving a GKE cluster validation for nodepools with same name does not happen

**To Reproduce**

- Deploy a GKE v2 hosted cluster.
- Have two nodepools with same name.
- Create/Save cluster details
- No error is seen during creation.
- Error seen on cluster after deployment: `Cluster [c-mf29m] cannot have multiple nodepools with name <nane>`
- Edit cluster, user does not have access to edit the nodepool name, but has to delete one and add another one.

**Expected Result**
When creating with same nodepool names, Error should be seen and not allow user to save details

**Comment by Jono-SUSE-Rancher** [comment]
@vincent99 - Is this something where the front end should be doing validation prior to the user clicking to save/create the cluster details, or is this something where the API should toss an error message back indicating that duplicate nodepools with identical names is probably a no-no?

**Comment by vincent99** [comment]
I don't think we attempt to do much server-side validation to replicate what google is going to say, so probably just frontend.
